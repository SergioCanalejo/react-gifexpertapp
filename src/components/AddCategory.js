import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const handleInput = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const newCategory = inputValue.trim();
    if (newCategory.length) {
      setCategories((categories) => [...new Set([newCategory, ...categories])]);
    }
    setInputValue("");
  };

  const [inputValue, setInputValue] = useState("");
  return (
    <form onSubmit={handleSubmit}>
      <h2>Add Category</h2>
      <input type="text" value={inputValue} onChange={handleInput} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
