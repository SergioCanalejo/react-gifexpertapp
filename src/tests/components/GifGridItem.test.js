import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Testing GifGridItem component', () => {
  const props = {
    title: 'Dummy title',
    url: 'https://localhost/dummy.jpg',
  };
  const wrapper = shallow(<GifGridItem {...props} />);

  GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  };
  test('Must show the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Must be a p with the title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(props.title);
  });

  test('Must have the img like the props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(props.url);
    expect(img.prop('alt')).toBe(props.title);
  });

  test('Must have animate__fadeIn', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('animate__fadeIn')).toBeTruthy();
  });
});
