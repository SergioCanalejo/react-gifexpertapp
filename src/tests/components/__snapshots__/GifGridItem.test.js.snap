// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem component Must show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Dummy title"
    src="https://localhost/dummy.jpg"
  />
  <p>
    Dummy title
  </p>
</div>
`;
